docs/JavaCodeQuality.md

```diff
@@ -521,6 +521,58 @@ private static final int SIXTIES_START = 1960; // good
 }
 ````
 
+### M02
+
+**Observation: @SuppressWarnings is used**  
+**Problem:** The suppressed rule detects problems, suppressing the detected violations without full knowledge can lead to the problems the rule is trying to prevent.  
+**Solution:** Only suppress violations (warnings) when you have complete understanding of the rule, and you are sure it does not apply.
+When suppressing warnings, document your reasoning and report false positives to the rule maintainers to help improve rule accuracy.  
+**Rule name:** UsingSuppressWarnings  
+**Example:**
+```java
+@SuppressWarnings("pmd:UnconditionalCreatedLogArguments") // Reason for suppression: the rule doesn't recognize record getters (yet), and those are not expensive
```

Comment:
ik geloof dat je ook **PMD**.UnconditionalCreatedLogArguments kunt doen, maar dat dat alleen in PMD werkt. Misschien even checken en ook toevoegen aan doc.

Reply:
hmm interesting: `PMD.` works in Sonar and IntelliJ PMD plugin, `pmd:` only works in Sonar!

Reply:
Can it trigger this regex? 

```xml
            <property name="ruleIdMatches" type="String" value="AvoidUserDataInSharedObjects|AvoidUnguardedMutableFieldsInSharedObjects|AvoidUnguardedAssignmentToNonFinalFieldsInSharedObjects|AvoidMutableStaticFields|[^\w]ALL[^\w]|[^\w]all[^\w]|PMD[^\.]|pmd[^:]"
                      description="Regex for inclusion of high risk rules"/>
```

Reply:
note that when using `PMD.` there is also a list of "Suppressed violations by Annotation" in IntelliJ plugin report

Reply:
ah, I didn't know about `pmd.`. Right, in the regex it is not one of those.

Reply:
I didn't know `PMD.` also works in sonar. So, that would be the best way, right? both tools supported, PMD shows suppressedViolations. Does it not do that with `pmd:`?

Reply:
no, there is no "suppressed violations" with `pmd:`, and also the suppress itself does not work in PMD... there is only the "UsingSuppressWarning"... so `PMD.` seems the best option now

Reply:
what is reason of `PMD[^\.]|pmd[^:]`? -> seems: using prefix of PMD or pmd without . or : respectively will trigger the `UsingSuppressWarningsHighRisk` rule? why?

Reply:
No, de `[^..]` means a `**not** any of the characters`

Reply:
Ah, found it in unit test, and added this description: 

The `@SuppressWarnings("pmd")` or `@SuppressWarnings("PMD")` are considered high risk as it suppresses many rules at once, use explicit rule list instead.

Reply:
should we fix the UsingSuppressWarningsHighRisk regex to not match on `pmd.` and `PMD:`?

Reply:
Nice to have documentation in the unit tests :-)

Reply:
Yes, I would think so. Let's see how it works out.